### DISCUSSION TITLE
opengothic-v1.0.2207

### DISCUSSION THREAD

**Discussion by Try** [Try/OpenGothic#474] [General]
![изображение](https://github.com/Try/OpenGothic/assets/122894/c921a4e1-99ee-45dd-ab32-5c338077de6e)

|   |  |
|---|-------|
| ![изображение](https://github.com/Try/OpenGothic/assets/122894/395455cc-c8b3-4186-a70f-39ae0d0b2c02)  | ![изображение](https://github.com/Try/OpenGothic/assets/122894/10fe2ef4-2de1-4452-8509-fe2c68c135ee) | 
![изображение](https://github.com/Try/OpenGothic/assets/122894/e90e3b2e-c84c-4e71-8a96-831520755b22) | ![изображение](https://github.com/Try/OpenGothic/assets/122894/3b3af162-e7a4-41c7-bdd8-15028b6cfe7e) | 
![изображение](https://github.com/Try/OpenGothic/assets/122894/8dce548f-71cd-40f8-b51f-e4a752f23bb9) | ![изображение](https://github.com/Try/OpenGothic/assets/122894/3a3cc343-d858-4699-86a1-db7b545c23d0)

* Game
  * Action focus is no longer available, if player is unconscious/dead
  * Improve camera inertia
  * AI now can use ladder for path-finding 
  * Ice-block spell: auto close inventory, if player is unconscious
  * Fix ice-block stan duration
  * Fix sell-price, when selling more than one item (#429)
  * Disallow turning animation when action-key is held down. (#400)
  * Disallow underwater way-points in AI_Flee
  * Implement sneaking reactions
  * Rework damage calculation (#434)
  * Remove room check for SENSE_HEAR
  * Align spell-cast player controls to original game
  
* General
  * Optimize save-game time
  * Added rechargeable spells support
  * Fix Greg's hat display
  * Biff dialogs: add orderId's for aiStopProcessInfo
  * Added fallback for invalid start point (#443)
  * Fix input problem when releasing one of two mapped keys (#441)
  * GothicStarter: add -devmode flag support
  * Add zCVobStair object type
  * Add gothic1 skill values (#423)
  * Added extra marvin commands (#398, #404, #471)
  * Change waypoint indexing to support G1

* Graphics
  * Physical lighting support
  * Add irradiance support
  * RayTracing: use hint gl_RayFlagsCullNoOpaqueEXT, if bindless in not available
  * RayTracing: use proper opaque flags in TLAS
  * RayTracing: enable bindless textures in DX12
  * RayTracing: improve bindless for Vulkan
  * Added stochastic fog support
  * Optimize fog for low-quality setups
  * Rescale alpha, after a-test
  * Added blendOut for animations
  * Better support for large-coordinates
  * Water wave rendering
  * Togglable water reflections
  * Brightness/Contrast/Gamma settings
  * Optimize fog shading
  * Optimize meshlet packaging
  
* MacOS 
  * Automatic installation detection
  * Fix key input in save-game menu
  * Query max texture size on Metal
  * Cleanup and generalize work with MacOS coordinate system
  * Fix buffer alignment issue in Metal
  * Make Mac-Command keys actionable (#402)
  
* Script-api
  * Align PERC_ASSESSPLAYER to vanilla
  * Add DAM_CRITICAL_MULTIPLIER, for Gothic 1 
  * Implemented AI_LOOKAT (#397)
  * wld_insertnpc: allow empty waty-point names
  
* Bugs:
  * Fix hnpc stashing at transform spells
  * Fix PfxController in oldworld.zen
  * Fix AI_UseItem
  * Fix issue, when using transform spell, that is last in inventory
  * Fix for G1 chests (#422)
  * Fix npc deduction in wld_playeffect
  * Prevent ranged damage from hitting twice (#424)
  * Avoid issue in Silvio quest with Npc_ExchangeRoutine
  * Avoid crash with Union.vdf
  * Avoid soft-lock in B_AssessQuietSound 
  * npc_isdead : return false if npc is null (#457)
  * Adjust camera selection for G1 (#437)
  * mdl_applyoverlaymdstimed: do not apply overlay, if not compatible

<hr /><em>This discussion was created from the release <a href='https://github.com/Try/OpenGothic/releases/tag/opengothic-v1.0.2207'>opengothic-v1.0.2207</a>.</em>

**Comment by apan65** [comment]
Temporal AA would be nice, or smaa shader.

**Reply by Try** [reply]
Yes, I'm researching topic in background;
`TAA` - not great, since we have no velocity map, what will cause ghosting.
`TAA` + `VK_NV_optical_flow` nice but require latest NV gpu with AI support
`FXAA` and probably also `SMAA` - not promising as they trying to do smart blur essentially, without addressing loss of signal
`MSAA` - not friendly to deferred rendering
`MSAA` + `VK_EXT_multisampled_render_to_single_sampled` - promising
`VK_NV_fragment_coverage_to_color` - also interesting, allowing to render GBuffer almost as usual, and have a side channel with coverage mask

**Reply by apan65** [reply]
DLAA for nvidia, FSR2\TAA for amd. Other methods are outdated

**Reply by Nindaleth** [reply]
FSR2 would be cool as a vendor-agnostic open-source upscaling method for both nVidia and AMD, also for Intel and possibly Apple/Broadcom/etc. GPUs, but is FSR2 really viable to use for 1:1 "scaling" to provide that velocity map to TAA even when no upscaling takes place?

**Comment by toolatetotheparty** [comment]
This is fire!

**Reply by Nindaleth** [reply]
+1, congratz to a beautiful release!

**Comment by inter-arc** [comment]
I was already starting to worry, no updates for quite some time.
But now I can clearly see that you have been really busy, thank you very much for your work!

**Comment by z0n4r1e** [comment]
this is amazing! if you will finish your project, it will be perfect.
